lib/vyasa_web/live/source_live/chapter/index.ex

```diff
@@ -48,28 +47,30 @@ defmodule VyasaWeb.SourceLive.Chapter.Index do
   defp sync_session(socket), do: socket
 
   defp apply_action(socket, :index, %{"source_title" => source_title, "chap_no" => chap_no} = _params) do
-    chap  = %{verses: verses, translations: [ts | _]} = Written.get_chapter(chap_no, source_title, @default_lang)
-
-    socket
-    |> stream(:verses, verses)
-    |> assign(:source_title, source_title)
-    |> assign(:chap, chap)
-    |> assign(:selected_transl, ts)
-    |> assign_meta()
+    with %Source{id: sid} = source <- Written.get_source_by_title(source_title),
+         %{verses: verses, translations: [ts | _]} = chap  <- Written.get_chapter(chap_no, sid, @default_lang) do
+
+      socket
+      |> stream(:verses, verses)
+      |> assign(:src, source)
+      |> assign(:chap, chap)
+      |> assign(:selected_transl, ts)
```

Comment:
(4) Correct to say that the scope of variables within the chained generators can be accessed from within it's following code-block? 

so here, the chained generators would be: 
* `%Source{id: sid} = source <- Written.get_source_by_title(source_title)` ==> so `sid` can be used by the following generator
* `%{verses: verses, translations: [ts | _]} = chap <- Written.get_chapter(chap_no, sid, @default_lang)`

and whatever is on the LHS of the generator clauses are being accessed from within the following code-block

Reply:
answered in prev comment, yes to this